backend/src/cache/storage/execution_cache_store.go

```diff
@@ -58,7 +58,7 @@ func (s *ExecutionCacheStore) GetExecutionCache(executionCacheKey string, maxCac
 	return latestCache, nil
 }
 
-func (s *ExecutionCacheStore) scanRows(rows *sql.Rows, podMaxCacheStaleness int64) ([]*model.ExecutionCache, error) {
+func (s *ExecutionCacheStore) scanRows(rows *sql.Rows, podCacheStaleness int64, maximumCacheStaleness int64) ([]*model.ExecutionCache, error) {
```

Comment:
I think `scanRows()` is intended for reading from the DB, but you are adding an additional functionality to also remove the stale cache. Could you separate the removal into an individual function?